src/Microsoft.AspNetCore.Razor.Language/Legacy/TagHelperParseTreeRewriter.cs

```diff
@@ -488,7 +488,13 @@ private bool HasAllowedChildren()
 
         private void ValidateParentAllowsContent(Span child, ErrorSink errorSink)
         {
-            if (HasAllowedChildren())
+            var isDisallowedContent = true;
+            if (_featureFlags.AllowHtmlCommentsInTagHelpers)
```

Comment:
Do this bit inside the `HasAllowedChildren()` block to avoid doing the work if unneeded.

Reply:
There is logical difference between those two checks. HasAllowedChildren has a different meaning, so it doesn't make sense to combine those. I'll just move that into an if block.